api/spec/api_docs/definitions/resources/stylesheet.rb

```diff
@@ -0,0 +1,16 @@
+module ApiDocs
+  module Definitions
+    module Resources
+      class Stylesheet
+
+        REQUIRED_CREATE_ATTRIBUTES = [:name].freeze
+
+        READ_ONLY_RELATIONSHIPS = [:text].freeze
```

Comment:
What do you think of this as an interface for noting which relationships are read only (and consequently which ones are writable)? Eventually, I think we should include writable relationships in the request examples on the documentation, which means that the `ApiDocs::Definitions::Resource` module and the `ApiDocs::Helpers::Request` module will need to know which relationships are not writable.

This makes an assumption that every route that has included resources attached to it in it's response, will be able to write to those relationships that are allowed through the validator. Is this a correct assumption?

For example, a stylesheet has included `text` and `text_sections` relationships, so if I pass in a `READ_ONLY_RELATIONSHIPS` value of `text`, then the `text_sections` relationships will end up as part of an example request on the documentation.

Reply:
I think this is OK for now. Some inconsistency will creep in because we're effectively copying logic from the validation concern into the API definition, so I expect this will change. With this type system we can introspect our serializers and understand what they're returning. We don't have a similar system for the request side of things, as the question of what is writable is answered in a rather ad hoc way in that validation concern. Adding this interface to the API definitions is OK for now, and as far as I'm concerned whether we add it as READ_ONLY_RELATIONSHIPS or WRITABLE_RELATIONSHIPS is a six of one half dozen of another question.